presto-common/src/main/java/com/facebook/presto/common/block/SingleMapBlock.java

```diff
@@ -171,12 +172,27 @@ int[] getHashTable()
     /**
      * @return position of the value under {@code nativeValue} key. -1 when key is not found.
      */
-    public int seekKey(Object nativeValue, MethodHandle keyNativeHashCode, MethodHandle keyBlockNativeEquals, MethodHandle keyBlockHashCode)
+    public int seekKey(Object nativeValue, MethodHandle keyNativeHashCode, MethodHandle keyBlockNativeEquals, MethodHandle keyBlockHashCode, int mapLookupsWithoutHashTable)
     {
         if (positionCount == 0) {
             return -1;
         }
 
+        mapLookups += 1;
+        if (mapLookups <= mapLookupsWithoutHashTable) {
+            for (int i = 0; i < positionCount / 2; i++) {
+                try {
+                    if ((Boolean) keyBlockNativeEquals.invoke(mapBlock.getRawKeyBlock(), offset / 2 + i, nativeValue)) {
+                        return 2 * i + 1;
+                    }
+                }
+                catch (Throwable throwable) {
```

Comment:
Per Effective Java, you should almost never catch Throwable. What's the more specific exception here?